src/librustc_middle/mir/mod.rs

```diff
@@ -1075,6 +1075,8 @@ pub enum TerminatorKind<'tcx> {
         discr: Operand<'tcx>,
 
         /// The type of value being tested.
+        /// This is always the same as the type of `discr`.
+        /// FIXME: remove this redundant information. Currently, it is relied on by pretty-printing.
```

Comment:
If you want to be more aggressive about it you can `#[deprecate]` it.

Reply:
Not sure what that would help with?

Reply:
It would nudge people away from new usages of this field.